doc/how-to-package-haskell-code.rst

```diff
@@ -0,0 +1,291 @@
+mode. The interactive mode will walk you through many of the package
+options and metadata, the non-interactive mode will simply pick reasonable
+defaults which is sufficient if you're just trying something out.
+
+.. highlight:: console
+
+::
+
+    $ cabal init --non-interactive
+    # You can also use -n which is the short version of --non-interactive
+
+If you want, you can also try out the interactive mode, for now chose
+"Executable" when asked what type of package you want to build.
+
+.. highlight:: console
+
+::
+
+    $ cabal init
+    ...
+    What does the package build:
+       1) Executable
+       2) Library
+       3) Library and Executable
+       4) Test suite
+    Your choice?
+
+One of the important questions is whether the package contains a library
+and/or an executable. Libraries are collections of Haskell modules that
+can be re-used by other Haskell libraries and programs, while executables
+are standalone programs. Test suites can both depend on a library or be
+standalonely generated.
+
+For the moment these are the only choices. For more complex packages
+(e.g. a library and multiple executables) the ``.cabal``
+file can be edited afterwards.
+
+After you make your selection (executable; library; library
+and executable; or: test suite) cabal asks us a number of questions starting with
+which version of the cabal specification to use, our package's name
+(for example, "proglet"), and our package's version.
+
+::
+
+    Generating CHANGELOG.md...
+    Generating Main.hs...
+    Generating proglet.cabal...
+
+Use the ``ls`` command to see the created files:
+
+::
+
+   $ ls
+   CHANGELOG.md  Main.hs  proglet.cabal
+
+
+Running the program
+-------------------
+
+Now that we have our Haskell code and the extra files that Cabal needs we
+can build and run our application.
+
+::
+
+   $ cabal build
+   Resolving dependencies...
+   ...
+   Linking /path/to/proglet ...
+
+   $ cabal run proglet
+   ...
+   Hello, Haskell!
+
+Since we have an executable we can use ``cabal run proglet`` which will build
+our executable (and re-build it if we've made any changes) and then run the
+binary. The ``cabal run`` command works for any ``component-name`` (tests for
+example), not just the main executable.
+
+
+About the Cabal package structure
+---------------------------------
+
+It is assumed that all the files that make up a package live under a common
+root directory (apart from external dependencies). This simple example has
+all the package files in one directory, but most packages use one or more
+subdirectories.
+
+Cabal needs one extra file in the package's root directory:
+
+-  ``proglet.cabal``: contains package metadata and build information.
+
+
+Editing the .cabal file
+-----------------------
+
+.. highlight:: cabal
+
+Load up the ``.cabal`` file in a text editor. The first part of the
+``.cabal`` file has the package metadata and towards the end of the file
```

Comment:
```suggestion
``.cabal`` file has the package metadata, and towards the end of the file
```
You may alos prefer to say "followed by" instead of "towards the end of the file".